app/assets/javascripts/components/common/wiki_select.jsx

```diff
@@ -11,167 +10,149 @@ import { formatOption } from '../../utils/wiki_utils';
-  },
-
-  render() {
-    if (this.props.readOnly) {
-      const lastIndex = this.props.wikis.length - 1;
-
-      const wikiList = map(this.props.wikis, (wiki, index) => {
-        const comma = (index !== lastIndex) ? ', ' : '';
-        const wikiUrl = this.url(wiki);
-        return <span key={wikiUrl}>{wikiUrl}{comma}</span>;
-      });
-      return (
-        <>
-          {wikiList}
-        </>
-      );
-    }
+  }, []);
+
+  if (readOnly) {
+    const lastIndex = wikis.length - 1;
+    const wikiList = map(wikis, (wiki, index) => {
+      const comma = (index !== lastIndex) ? ', ' : '';
+      const wikiUrl = url(wiki);
+      return <span key={wikiUrl}>{wikiUrl}{comma}</span>;
+    });
+
+    return (
+      <>
+        {wikiList}
+      </>
+    );
+  }
 
-    // Used to set the already available wikis
-    let wikis = [];
-    if (this.props.wikis) {
-      wikis = this.props.wikis.map((wiki) => {
-        wiki.language = wiki.language || 'www'; // for multilingual wikis language is null
-        return this.formatOption(wiki);
-      });
-    }
+  // Used to set the already available wikis
+  let formattedWikis = [];
+  if (wikis) {
+    formattedWikis = wikis.map((wiki) => {
+      wiki.language = wiki.language || 'www'; // for multilingual wikis language is null
+      return formatOption(wiki);
+    });
+  }
 
-    // Home Wiki should appear first in the list of tracked wikis as in any other place it blocks
-    // the removal of a wiki via backspace. Removing a wiki by pressing 'X' still works
-    // but is bad UI and home wiki appearing first makes more sense anyway.
-    let home_wiki = this.props.homeWiki;
-    if (this.props.multi) {
-      home_wiki = this.formatOption(home_wiki);
-      wikis = ArrayUtils.removeObject(wikis, home_wiki);
-      wikis.unshift(home_wiki);
-    }
+  // Home Wiki should appear first in the list of tracked wikis
+  let formattedHomeWiki = homeWiki;
+  if (multi) {
+    formattedHomeWiki = formatOption(homeWiki);
+    formattedWikis = ArrayUtils.removeObject(formattedWikis, formattedHomeWiki);
+    formattedWikis.unshift(formattedHomeWiki);
+  }
 
-    const preprocess = (wiki) => {
-      if (wiki) {
-        if (this.props.multi) {
-          wiki = wiki.map((w) => { return { value: JSON.parse(w.value), label: w.label }; });
-        } else {
-          const value = JSON.parse(wiki.value);
-          wiki = { label: wiki.label, value };
-        }
-        this.props.onChange(wiki);
+  const preprocess = (wiki) => {
+    if (wiki) {
+      if (multi) {
+        wiki = wiki.map((w) => { return { value: JSON.parse(w.value), label: w.label }; });
       } else {
-        this.props.onChange(this.props.multi ? [] : {});
+        const value = JSON.parse(wiki.value);
+        wiki = { label: wiki.label, value };
       }
-    };
-
-    // If the component was passed a set of wikis as options, limit the options to those wikis.
-    // Otherwise, allow any wiki to be selected.
```

Comment:
What is the reason for not retaining these comments?

Reply:
Hey @ragesoss sorry for overlooking those comments I have updated my code.